src/main/java/com/cleanroommc/modularui/widgets/DropDownMenu.java

```diff
@@ -218,5 +201,114 @@ public DropDownWrapper background(IDrawable... background) {
             }
             return super.background();
         }
+
+        private void rebuild() {
+            if (!isValid()) return;
+            Area parentArea = getParent().getArea();
+            size(parentArea.width, parentArea.height * maxItemsOnDisplay);
+            pos(0, direction == DropDownDirection.UP ? parentArea.height * (maxItemsOnDisplay + 1) : parentArea.height);
+
+            List<IWidget> children = getChildren();
+            for (int i = 0; i < children.size(); i++) {
+                IWidget child = children.get(i);
+                child.getArea().setSize(parentArea.width, parentArea.height);
+                child.getFlex().left(0).top(parentArea.height * i);
+                child.setEnabled(open);
+            }
+        }
+    }
+
+    public static class DropDownItem extends Widget<DropDownItem> implements Interactable {
```

Comment:
Aren't these implementations just copy of `ButtonWidget`? How about `class DropDownItem extends ButtonWidget<DropDownItem>` and remove these boilerplates?